artifact/sha/sha.hpp

```diff
@@ -58,6 +60,43 @@ extern const ErrorCategoryClass KeyValueParserErrorCategory;
 
 error::Error MakeError(ErrorCode code, const string &msg);
 
+class SHA {
+public:
+	SHA() :
+		sha_ {} {};
+	SHA(vector<uint8_t> &v) :
+		sha_ {v} {};
+	operator string() const {
```

Comment:
Especially since this does conversion, I would suggest using a dedicated `String()` function instead of automatic casting. We've already seen how automatic conversion can go wrong, so it might be wise to use it sparingly.

Reply:
Hmm, but in this case I want the object to be transparently dual.

I want it to cast to `string` when required, and `vector` when required.

Reply:
But why do you need it to be transparent? They don't even have the same content. The string conversion converts it to a human-readable string, so it's not really the same object anymore. I think this should be explicit.

Your `string(variable)` would just change to `variable.String()`, but it has the benefit of warning you if you start mixing types (which you probably don't want).

Reply:
I liked it tbh, but it's fine. both work. I give